### PR TITLE
[dominoswiss] Initial contribution

### PR DESCRIPTION
**Pull request by Friesoch**
Added new Binding "org.openhab.binding.dominoswiss" for a first initial commit of this binding

[dominoswiss][WIP] initial contribution

added new binding org.openhab.binding.dominoswiss for initial commit
This binding allows the controll of rollershutters, using an egate as gateway and dominoswiss radio receivers. The egate-gateway is connectet via ethernet to openhab and sends it's commands via radio to all rollershutters.

Announcement: https://community.openhab.org/t/announce-dominoswiss-binding/76503
Replaces Pull-Request https://github.com/openhab/openhab-addons/pull/9837

Jar-File for Testing:
https://github.com/Friesoch/org.openhab.binding.dominoswiss

### CODE REVIEW

**Comment by Friesoch** [comment]
@fwolter here is the new Pull Request replacing the corrupted https://github.com/openhab/openhab-addons/pull/9837

File: bom/openhab-addons/pom.xml [hunk 1]
```diff
@@ -321,6 +321,11 @@
       <artifactId>org.openhab.binding.dmx</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+    <groupId>org.openhab.addons.bundles</groupId>
+      <artifactId>org.openhab.binding.ddominoswiss</artifactId>
```
Comment (wborn) [inline bom/openhab-addons/pom.xml:326]: This typo may cause your PR build to fail.

```suggestion
      <artifactId>org.openhab.binding.dominoswiss</artifactId>
```

File: bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/EGateHandler.java [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+    private void pollingConfig() {
+        if (!isConnected()) {
+            Socket localSocket = egateSocket;
+
+            synchronized (lock) {
+                try {
+                    localSocket = new Socket();
+                    localSocket.connect(new InetSocketAddress(host, port));
+                    localSocket.setSoTimeout(SOCKET_TIMEOUT_SEC);
+                    BufferedWriter localWriter = writer;
```
Comment (fwolter) [inline bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/EGateHandler.java:220]: This assignment has no effect as it is overwritten in the next line.

File: bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/EGateHandler.java [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+                    localWriter = new BufferedWriter(new OutputStreamWriter(localSocket.getOutputStream()));
```
Comment (fwolter) [inline bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/EGateHandler.java:221]: You need to close the BufferedWriter. You could use the auto-close feature:
```suggestion
                    try(localWriter = new BufferedWriter(new OutputStreamWriter(localSocket.getOutputStream()))) {
```

If I see correctly, you create new BufferedWriter/OutputStreamWriter each poll. Why not create one when connecting and close it in dispose() or disconnect? Then, you can also re-use it in sendCommand().

**Review by fwolter — CHANGES_REQUESTED** [review]
There's one checkstyle warning left.

File: bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/EGateHandler.java (hunk 3 above)
Comment (Friesoch) [reply bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/EGateHandler.java:220]: I close the BufferedWriter. The redesign will be made with a later release, thanks for that hint.

Comment (fwolter) [reply bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/EGateHandler.java:220]: The BufferedWriter is assigned to the local variable `localWriter`, but this variable is out of scope if the method `pollingConfig` is left. So, the BufferedWrite is never closed and you will end up with a resource leak.

As said, the assignment `BufferedWriter localWriter = writer;` has no effect.

File: bom/openhab-addons/pom.xml [hunk 4]
```diff
@@ -321,6 +321,11 @@
       <artifactId>org.openhab.binding.dmx</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+    <groupId>org.openhab.addons.bundles</groupId>
+      <artifactId>org.openhab.binding.dominoswiss</artifactId>
```
Comment (fwolter) [inline bom/openhab-addons/pom.xml:326]:
```suggestion
    <dependency>
      <groupId>org.openhab.addons.bundles</groupId>
      <artifactId>org.openhab.binding.dominoswiss</artifactId>
```

**Review by fwolter — CHANGES_REQUESTED** [review]

**Review by fwolter — APPROVED** [review]
LGTM

Now, you could add your binding's logo to the openHAB website. See https://www.openhab.org/docs/developer/bindings/#add-your-binding-s-logo-to-the-openhab-website

**Comment by Friesoch** [comment]
> LGTM
> 
> Now, you could add your binding's logo to the openHAB website. See https://www.openhab.org/docs/developer/bindings/#add-your-binding-s-logo-to-the-openhab-website

Dear @fwolter thanks a lot for your support and your input to bring the binding to the desired quality level :) A big thank you and cheers

**Pull request merged**